File: app-server/app-server/Domain/Entities/Reservation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace app_server.Domain.Entities;

[Table("reservations")]
public class Reservation
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int id { get; set; }
    [Required]
    public int userId { get; set; }
    [Required]
    public int spaceId { get; set; }
    [Required]
    public DateTime startDate { get; set; }
    [Required]
    public DateTime endDate { get; set; }
    
    public virtual User user { get; set; }
    
    public virtual Space space { get; set; }
}

File: web-app/src/app/shared/types/reservation.ts
import {User} from './user';
import {Space} from './space';

export interface Reservation {
  id: number;
  userId: string;
  spaceId: number;
  startDate: Date;
  endDate: Date;
  user: User;
  space: Space;
}

export interface ReservationParams {
  userId?: number;
  spaceId?: number;
  startDate?: Date;
  endDate?: Date;
  fetchPlaces?: boolean;
}


export interface CreateReservation {
  userId: number;
  spaceId: number;
  startDate: Date;
  endDate: Date;
}
